lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -0,0 +1,393 @@
+require 'digest/md5'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PayuLatamGateway < Gateway
+      self.display_name = "PayU Latam"
+      self.homepage_url = "http://www.payulatam.com"
+
+      self.test_url = "https://sandbox.api.payulatam.com/payments-api/4.0/service.cgi"
+      self.live_url = "https://api.payulatam.com/payments-api/4.0/service.cgi"
+
+      self.supported_countries = ["AR", "BR", "CO", "MX", "PA", "PE"]
+      self.default_currency = "USD"
+      self.money_format = :dollars
+      self.supported_cardtypes = [:visa, :master, :american_express, :diners_club]
+
+      BRAND_MAP = {
+        "visa" => "VISA",
+        "master" => "MASTERCARD",
+        "american_express" => "AMEX",
+        "diners_club" => "DINERS"
+      }
+
+      def initialize(options={})
+        requires!(options, :merchant_id, :account_id, :api_login, :api_key)
+        super
+      end
+
+      def purchase(amount, payment_method, options={})
+        post = {}
+
+        add_credentials(post, 'SUBMIT_TRANSACTION')
+        add_transaction_type(post, 'AUTHORIZATION_AND_CAPTURE')
+        add_order(post, options)
+        add_buyer(post, options)
+        add_invoice(post, amount, options)
+        add_signature(post)
+        add_payment_method(post, payment_method)
+        add_payer(post, options)
+        add_browser_data(post, options)
+        add_extra_parameters(post, options)
+
+        commit('purchase', post)
+      end
+
+      def authorize(amount, payment_method, options={})
+        post = {}
+
+        add_credentials(post, 'SUBMIT_TRANSACTION')
+        add_transaction_type(post, 'AUTHORIZATION')
+        add_order(post, options)
+        add_buyer(post, options)
+        add_invoice(post, amount, options)
+        add_signature(post)
+        add_payment_method(post, payment_method)
+        add_payer(post, options)
+        add_browser_data(post, options)
+        add_extra_parameters(post, options)
+
+        commit('auth', post)
+      end
+
+      def capture(authorization, options={})
+        post = {}
+
+        add_credentials(post, 'SUBMIT_TRANSACTION')
+        add_transaction_type(post, 'CAPTURE')
+        add_reference(post, authorization)
+
+        commit('capture', post)
+      end
+
+      def void(authorization, options={})
+        post = {}
+
+        action, _, _ = split_authorization(authorization)
+
+        case action
+        when 'auth'
+          add_credentials(post, 'SUBMIT_TRANSACTION')
+          add_transaction_type(post, 'VOID')
+          add_reference(post, authorization)
+        when 'purchase', 'capture'
```

Comment:
Usually `void` happens on authorizations and `refund` happens on captures and purchases.  I'm wondering if we can have a `def refund` and split it out into 2 operations.